src/config/resource.hpp

```diff
@@ -9,7 +9,7 @@
 #include <string>
 #include <unordered_map>
 
-typedef std::unordered_map<std::string, ProtoType*> AttributeMap;
+typedef std::shared_ptr<std::unordered_map<std::string, std::shared_ptr<ProtoType>>> AttributeMap;
```

Comment:
There are a lot of places where we use this exact signature, instead of the `typedef` here. Might make sense to try and replace those longer, annoying signatures with `AttributeMap`. 

Note that this _might_ get you into some circular dependency issues that might be pretty annoying. If that happens then feel free to just skip!

Reply:
was causing errors in a few files but changed a lot of them to AttributeMap!